### ISSUE TITLE
Failure in starting container with "docker run -it ubuntu:18.04 bash"

### ISSUE THREAD

**Issue by michael2012z** [lf-edge/runx#2]
Hi, @sstabellini 

I tested runx on a X86 host, and run into a failure in starting the VM.

The environment is:
- The host is Ubuntu 20.04 with kernel version 5.4.0.
- Xen was installed via apt, not from source code. Version is 4.11.

I changed dockerd runtime option to work with runx:
> usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock --add-runtime runx-runtime=/usr/sbin/runX --default-runtime=runx-runtime

I tried to start a ubuntu container with interactive bash with:
> docker run -it ubuntu:18.04 bash

Guest kernel began to load but failed soon, 9pfs was not mounted for rootfs, the error message was:
> [    1.897215] VFS: Cannot open root device "9p" or unknown-block(0,0): error -6

I have attached the complete guest kernel log and vm cfg. Could you help have a look and give me some help?

[runx-vm.log](https://github.com/lf-edge/runx/files/5358255/runx-vm.log)
[xen_vm.cfg.txt](https://github.com/lf-edge/runx/files/5358288/xen_vm.cfg.txt)

**Comment by sstabellini** [comment]
Thanks for the logs. I bet the issue is that Ubuntu didn't enable Xen 9pfs support in their qemu-xen build. Thus, 9pfs doesn't get exported to the VM correctly.

If I am right, you wouldn't need to rebuild the Xen hypervisor and tools, just QEMU: the binary that gets installed under /usr/lib/xen/bin/qemu-system-i386. See this script as an example on how to build it:

https://gitlab.com/ViryaOS/viryaos-xen/-/blob/master/main#L48

Note the --enable-virtfs parameter which is important, and likely it was not passed by Ubuntu package build.

**Comment by michael2012z** [comment]
Thank you for reply. 

I rebuilt QEMU with "--enable-virtfs" option and made sure /usr/lib/xen/bin/qemu-system-i386 points to the new binary. But the problem is same.

I captured the qemu command when the guest was starting:
`/usr/bin/qemu-system-i386 -xen-domid 62 -chardev socket,id=libxl-cmd,path=/var/run/xen/qmp-libxl-62,server,nowait -no-shutdown -mon chardev=libxl-cmd,mode=control -chardev socket,id=libxenstat-cmd,path=/var/run/xen/qmp-libxenstat-62,server,nowait -mon chardev=libxenstat-cmd,mode=control -nodefaults -no-user-config -xen-attach -name ecd8b2919f7e2728a7815fde8610a707c226a32f01e309195c3ba97c979d6f13 -vnc 127.0.0.1:0,to=99 -display none -machine xenpv -m 1024`

I googled about QEMU with 9pfs. Some posts mentioned that a 9p device should be added to QEMU with "-device" parameter, like https://lists.gnu.org/archive/html/qemu-devel/2016-09/msg07307.html. But I didn't have such option in my QEMU command. Was it the cause?

**Comment by sstabellini** [comment]
No, the QEMU command line shouldn't be the problem.

Could you please post the output of ```xenstore-ls``` after creating the VM when you get the 9pfs error?

Also, could you try to start a VM using ```xl create``` and the following config file:
```
name="test"
memory=1000
vcpus=1
kernel="/usr/share/runX/kernel"
ramdisk="/root/initrd.cpio"
extra="earlycon console=hvc0 root=/dev/ram0 rdinit=/bin/sh"

p9=[ 'tag=share_dir,security_model=none,path=/tmp' ]
```
Where initrd is just a trivial tiny busybox-based ramdisk. Also attached here as zipfile (need to unzip before using it)
[initrd.cpio.zip](https://github.com/lf-edge/runx/files/5373197/initrd.cpio.zip). You could rebuild one yourself using busybox-static and cpio.


Once the VM is started, you can try to mount the 9pfs filesystem with:



```
mount -t proc proc /proc
mount -t sysfs sysfs /sys
mount -t devtmpfs devtmpfs /dev
mount -t 9p -o trans=xen share_dir /mnt
```

The last mount command is the one that counts.

**Comment by michael2012z** [comment]
Thanks for your help.

Here comes the xenstore-ls output. I will try 9p manually with the commands you shared and come back with the result later.

[xenstore-ls.txt](https://github.com/lf-edge/runx/files/5375435/xenstore-ls.txt)

**Comment by michael2012z** [comment]
Hi, @sstabellini , the problem was resolved. Let me tell you the story. :)

I used the `initrd.cpio.zip` you attached to start a VM, 9p worked fine. I realized that something is wrong in my initrd. After double-checking everything, I recalled that in my first build, configuration file for X86 was missing in `initrd/` folder, there is only config.arm64. While I am testing on a X86 machine. So I moved to busybox folder and generated a default config with `make defconfig`, but problem is it hasn't `CONFIG_STATIC=y` configured. 

Just now I simply copied config.arm64 for config.x86, built and tested again. It worked.

Is the configuration file of busybox architecture specific? Maybe we can use the same config for both arch's.

Again, thank you very much for help.
